### ISSUE TITLE
🐛: When using 'open with' on HTML files it loads google.com page instead of the HTML file.

### ISSUE THREAD

**Issue by Kidcredo** [responsively-org/responsively-app#365]
![IwAKSPmTRP](https://user-images.githubusercontent.com/15522605/88883541-21720e80-d224-11ea-8883-51c6cabb05f6.gif)

**Comment by manojVivek** [comment]
That is strange.
It tries to loads that file and then goes back to google.com. 

@esprush Could this be because of the URL syncing? 

@jjavierdguezas Any ideas?

**Comment by jjavierdguezas** [comment]
I cant reproduce it, @Kidcredo can you please share the files?

**Comment by jjavierdguezas** [comment]
Thats weird @Kidcredo , it works in my PC like a charm

I completed the `fun.js` Here you have the used files: https://we.tl/t-rCG0FduozF

![openwithissue2](https://user-images.githubusercontent.com/13673443/88938316-8b6ed000-d285-11ea-904c-145b5b95aa24.gif)

**Comment by Kidcredo** [comment]
I am guessing this was because of some cache issue due to updating directly without uninstalling the 0.8.0 version first. 
Just uninstalled and did a clean install now it seems to be working fine.

**Comment by Kidcredo** [comment]
After the clean install it worked once and when I tried it again it happened again.
It happened again with the file you shared @jjavierdguezas 
![9DtPEXVKBx](https://user-images.githubusercontent.com/15522605/88943673-4d6cae80-d27b-11ea-9f98-558d0700c745.gif)

**Comment by jjavierdguezas** [comment]
Definitely something weird happens. I can see by myself that when you open a file using `Open with`, sometimes Responsively go a back to HOME_PAGE and the file, its like some race condition between redux state initialization and the open file logic. Somehow in my tests it always ends showing the desired file, but in your PC "got to home page" wins...

**Comment by jjavierdguezas** [comment]
I tried with multiple devices and in my PC it always works,

**Comment by manojVivek** [comment]
@jjavierdguezas @Kidcredo It is possible it could be because of the URL sync.

Possible explanation:
1. One of the devices in the preview is lagging in rendering during the app load.
2. When the `address-change` event is triggered. It is failed in the lagging device because the `webview` is not ready.
3. Finally, when the lagging device loads the default URL(google.com), the URL sync is transmitting that URL to all other devices.
4. Other devices process the event and load google.com as well.

If we watch the address bar URL changes closes, it confirms this behavior. 

cc: @esprush

**Comment by chris-53** [comment]
I am having the same issue. Watching this keenly in the hope someone finds a solution.

**Comment by Bobbbay** [comment]
Interestingly enough, I was half able to reproduce this issue - for myself, it opens up Google, then the file, then Google, then finally the file (successfully). 

Acer Swift 5, Windows

**Comment by jjavierdguezas** [comment]
@Kidcredo @manojVivek @Bobbbay @chris-53 I tested it many times and I think that after #586 this issue is fixed

**Comment by chris-53** [comment]
> 
> 
> @Kidcredo @manojVivek @Bobbbay @chris-53 I tested it many times and I think that after #586 this issue is fixed

it's still not working for me despite having updated itself.

**Comment by jjavierdguezas** [comment]
> > @jjavierdguezas @Kidcredo @manojVivek @Bobbbay @chris-53 I tested it many times and I think that after #586 this issue is fixed
> 
> it's still not working for me despite having updated itself.

It hasn't been released yet, try building yourself from master branch is you want to test it now

**Comment by chris-53** [comment]
> 
> 
> > > @jjavierdguezas @Kidcredo @manojVivek @Bobbbay @chris-53 I tested it many times and I think that after #586 this issue is fixed
> > 
> > 
> > it's still not working for me despite having updated itself.
> 
> It hasn't been released yet, try building yourself from master branch if you want to test it now

I only joined this forum to see what was going on. I don't build software, I am just a user. I'll wait until it gets fixed by whoever decides to fix it. meanwhile it's useless to me, so i don't use it. I won't hold my breath regarding seeing a proper fix as it has taken since last August for nothing to happen.

**Comment by Kidcredo** [comment]
Hey @chris-53 
I get where this is coming from but you need to understand that open source software costs time and resources of the people who are building and maintaining it. 
There is a popular saying in the open source community. If you want a feature the best way to get it is build it yourself.
If you cannot contribute through code you can contribute via funds and help this project move further. If the project has sufficient funds. Then maintainers can put in more time and efforts on this project and help ship new features and bug fixes faster.

If you chose not to use this app then the alternative you have is Sizzy. Which has a paid subscription model and software that's close sourced. 

Either way you end up paying in time or money.

**Comment by jjavierdguezas** [comment]
well, a new version was released, please try to test this issue after update. thanks

**Comment by chris-53** [comment]
Thanks @jjavierdguezas
Still no change unfortunately.

**Comment by jjavierdguezas** [comment]
> Thanks @jjavierdguezas
> Still no change unfortunately.

can you please share your environment? (Press `F1` in the app, clic the `Copy` button and paste here)

**Comment by chris-53** [comment]
@jjavierdguezas

Version: 0.17.0
Electron: 9.3.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19042

**Comment by jjavierdguezas** [comment]
> @jjavierdguezas
> 
> Version: 0.17.0
> Electron: 9.3.1
> Chrome: 83.0.4103.122
> Node.js: 12.14.1
> V8: 8.3.110.13-electron.0
> OS: Windows_NT x64 10.0.19042

I have the same env in one of my PCs... 

Could you share an example with which this error occurs to you? (an html file for example)

**Comment by chris-53** [comment]
@jjavierdguezas
I try to open an preview of a page in  web building software that I use so i am not sure what to send you that will help. Perhaps you could advise please? At the moment the browser opens Google as it has done since the first time I tried it.

**Comment by jjavierdguezas** [comment]
> @jjavierdguezas
> I try to open an preview of a page in web building software that I use so i am not sure what to send you that will help. Perhaps you could advise please? At the moment the browser opens Google as it has done since the first time I tried it.

I am assuming you go to a `PATH` in your filesystem, right click on a `FILE.html`, select the `Open With` option and finally click on `ResponsivelyApp` item. Right?
If so, we would need:
 - Analyse the PATH: if you can share it good, if not, at least tell us if that PATH has spaces, accented letters, symbols, etc
 - Analyse the FILE.html: if you can share it perfect (and all the files it uses like js or css files). You probably can't share it so it will be good if you can create an index.html with some shareable content which doesn't open in your environment following the above steps

I'm trying to reproduce your scenario as closely as possible.

**Comment by chris-53** [comment]
Thanks for trying all this. The programme I am using has a "preview" button which then opens a temporary file in the browser of my choice. Every other browser opens the temporary file. If I choose the Responsively browser it opens Google instead. If I open the file in another browser and then paste the file location into the Responsively browser it opens the file OK.
The location of the file on my computer in the page I have just tried to open is
file:///C:/Users/chist/AppData/Local/Temp/preview/index.shtml

**Comment by jjavierdguezas** [comment]
@chris-53 in this link: https://wormhole.app/VLoY1#v6I2KampyRngg9c7Qpo7-g
you can download two files: `index.html` and `fun.js` the same test scenario form https://github.com/responsively-org/responsively-app/issues/365#issuecomment-666401417

Could you download them, Open with Responsively and share the results?

**Comment by chris-53** [comment]
It opens and says Check the Console.

**Comment by jjavierdguezas** [comment]
> Thanks for trying all this. The programme I am using has a "preview" button which then opens a temporary file in the browser of my choice. Every other browser opens the temporary file. If I choose the Responsively browser it opens Google instead. If I open the file in another browser and then paste the file location into the Responsively browser it opens the file OK.
> The location of the file on my computer in the page I have just tried to open is
> file:///C:/Users/chist/AppData/Local/Temp/preview/index.shtml

Thanks for that information @chris-53 , here are my comments:
1- Responsively is not a browser, so maybe it is not fully compatible with the app that you are using
2- This issue is about opening `html` files with the 'Open with' option of Windows File Explorer. 
  - `shtml` is different from `html`, Responsively only opens `*.html` and `*.htm` files via "Open with" option. `*.shtml` files are meant for servers AFAIK. If I am wrong and you want Responsively to open those files, then please open a diferent issues.
  
As you can see, this issue is not your scenario
Any other request or bug report please open and comment in a new issue

**Comment by jjavierdguezas** [comment]
> It opens and says Check the Console.

So **this** issue does not happen to you

**Comment by jjavierdguezas** [comment]
@Kidcredo @manojVivek @Bobbbay If you can check this problem again in your environments it would be helpful. Thanks!

**Comment by chris-53** [comment]
Yes, you are right about shtml, it is a server side extension. My site used to use 'server side includes'. It no longer does but I have kept the extension because Google and numerous other sites link to the shtml files. It's a nuisance in this situation. I have one or two html files on the site and they open with Responsively OK. It looks as though I will have to give up on Responsively. Thanks for all you have done.

**Comment by Kidcredo** [comment]
I just tried it out seems to be working fine now.
